docs/api/v1.md

```diff
@@ -0,0 +1,117 @@
+# Spice API
+## Version `v1`
+
+All routes are prefixed with `/api/v1/`
+
+Example: `GET localhost:1234/api/v1/processes`
+
+## Host System Info
+
+### `GET /filesystem/:path*`
+Gets the file(s) within the given path
+
+Path parameters:
+ - `path` {string}: URL encoded path to a file or directory.  If empty, refers to the root directory.
+
+Responses:
+ - `200`:
+    - Returns {[File](#File)}
+ - `400`:
+    - Returns {[Error](#Error)}: Invalid path.
+ - `403`: 
+    - Returns {[Error](#Error)}: Insufficient privileges to open file.
+ - `404`: 
+    - Returns {[Error](#Error)}: File doesn't exit.
+
+### `GET /processes`
+Gets the list of processes running on the host machine.
+
+Responses:
+ - `200`:
+    - Returns {[Process](#Process)[]}
+ - `403`:
+    - Returns {[Error](#Error)}: Insufficient privileges to complete operation.
+
+## Debug
+All `/debug` routes are stateful--their responses are based on the currently attached process/executable.
+If the debugger is not attached, they respond with `40somethin`.
+
+### `POST /debug/attach/pid/:pid`
+Returns status of attached pid, execution id of current function.
+
+### `POST /debug/attach/bin/:path*`
+
+### `GET /debug`
+Returns information about the attached process or executable.
+
+### `GET /debug/functions`
+Returns a list of functions in the attached process.
+
+### `GET /debug/functions/:function`
+Returns source code and input fields for this function.
+
+### `PUT /debug/functions/:function/breakpoints`
+Sets a breakpoint on this function.
+
+### `DELETE /debug/functions/:function/breakpoints`
+Removes breakpoint on this function.
+
+### `GET /debug/breakpoints`
+Lists breakpoints.
+
+### `POST /debug/functions/:function/execute`
+Executes the function with parameters provided in POST body.
+Returns an [execution id](#execution).
+
+### `GET/debug/executions`
+Returns list of all executions.
+
+### `GET /debug/executions/:executionId`
+Returns execution status (function, processing/done, execution time).
+
+### `GET /debug/executions/:executionId/trace`
```

Comment:
This implies that the server has to keep a history of its execution traces. It might be nicer for low-power target machines if the client kept the history instead. I also don't know if we're going to be running more than one function at once. So, would it be possible to stream results back in response to a single `POST /debug/functions/:function/execute`?

I suppose on the other hand this makes it easier to keep HTTP while supporting multiple users (who could even be executing multiple functions at once, though that would be silly as they could just run them on their own instances - the benefit of multi-user would be for them all to see the same execution, I think).